compiler_opt/distributed/worker.py

```diff
@@ -30,16 +29,17 @@ def is_priority_method(cls, method_name: str) -> bool:
 
 
 # Dask's Futures are limited. This captures that.
-class WorkerFuture(Generic[T], metaclass=abc.ABCMeta):
+class WorkerFuture(Protocol[T]):
 
-  @abc.abstractmethod
   def result(self) -> T:
     raise NotImplementedError()
 
-  @abc.abstractmethod
   def done(self) -> bool:
     raise NotImplementedError()
 
+  def add_done_callback(self, fn) -> None:
```

Comment:
ugh... dask. OK, let's move forward with the change, though, we'll sort out Dask after (like we can have that configuration pay with a handler thread, for example)